src/heap.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+"""Testing module for the Heap classes."""
+
+class Heap(object):
+    """Heap class implementation"""
+    def __init__(self, min_max='max'):
+        """Initialize Heap"""
+        self._heap = []
+        self.size = 0
+        heap_types = ['max', 'min']
+        if min_max.lower() in heap_types:
+            self._min_max = min_max.lower()
+            if self._min_max == 'max':
+                self._shift_up = self._shift_up_max
+                self._shift_down = self._shift_down_max
+            else:
+                self._shift_up = self._shift_up_min
+                self._shift_down = self._shift_down_min
+        else:
+            raise TypeError("heap needs either 'min' or 'max'")
+
+    def insert(self, val):
+        """Insert a node into the heap."""
+        self._heap.append(val)
+        current_position = self.size
+        while current_position:
+            # import pdb; pdb.set_trace()
+            current_position = self._shift_up(current_position)
+        self.size += 1
```

Comment:
Out of curiosity, why did you choose to ignore the specification for this data structure and implement `insert` and `extract` rather than `push` and `pop`?